node/dataSources/catalog.ts

```diff
@@ -153,9 +154,15 @@ export class CatalogDataSource extends IODataSource {
     orderBy = '',
     from = 0,
     to = 9,
-    map = ''
+    map = '',
+    filterUnavailableBySegment = false,
   }: ProductsArgs) => {
     const sanitizedQuery = encodeURIComponent(decodeURIComponent(query).trim())
+    if (filterUnavailableBySegment) {
+      const segmentData: SegmentData | undefined = (this.context! as CustomIOContext).segment
```

Comment:
`(this.context! as CustomIOContext)` is this safe enough? will context always have a value?